src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseRepository.java

```diff
@@ -98,4 +103,12 @@ public Claim saveClaim(String authorisation, Claim claim) {
         return coreCaseDataService.save(authorisation, claim);
     }
 
+    @Override
+    public Optional<Claim> getByClaimReferenceNumberAnonymous(String reference) {
```

Comment:
Plase remove method from here. A `CCDCaseRepository` should not be responsible for authenticating anonymous user, that is responsibility of `uk.gov.hmcts.cmc.claimstore.services.ClaimService#getClaimByReferenceAnonymous`  method.